File: pascal/solve.pas
program Encrypt;

uses sysutils;

var
  input: string;
  numShift: integer;
  i: integer;

function shift(strInput: string; num: integer): string;
var
  i: integer;
  character: char;
begin
     if num >= 26 then
         num := num mod 26;
     for i := 1 to Length(strInput) do begin
         character := strInput[i];
         if((ord(character) - num) >= ord('A')) then
	      strInput[i] := chr(ord(character) - num)
	  else
	      strInput[i] := chr(ord('Z') - ((num - 1) - (ord(character) - ord('A'))))
     end;
     
     shift := strInput;
end;

begin
  writeln('Enter a string');
  readln(input);
  writeln('Enter a number');
  readln(numShift);
  
  input := upcase(input);
  
  for i := 1 to numShift do begin
      writeln('Caesar ' + IntToStr(numShift - i) + ': ' + shift(input, i));
  end;
end.

File: erlang/solve.erl
-module(solve).   
-export([solve/2]). 

shift(C, RealShift) ->
	if 		
		C+RealShift =< $Z -> C+RealShift;
		true -> $A + (C+RealShift - $Z) -1
	end.

encrypt(Input, NumShift) -> 
	RealShift = NumShift rem 26,
	lists:map(fun(C) -> shift(string:to_upper(C), RealShift) end, Input).	

solve(Input, 0) ->
 	io:format("~s~n", [string:concat("Caesar 0: ",string:to_upper(Input))]);
solve(Input, NumShift) when NumShift > 0 -> 
	io:format("~s~n", [lists:concat(["Caesar ", NumShift, ": ", encrypt(Input, NumShift)])]),
	solve(Input, NumShift-1).
